Automattic/jetpack
# Protect: Add WAF post upgrade messaging

#### Description
Adds post upgrade messaging to the `FirewallPage`

#### Changes proposed in this Pull Request:

* Adds an upgrade `Popover` component above the Automatic rules toggle
* Adds an upgrade badge beside the Automatic rules text
* Adds a `/waf-upgrade-seen` endpoint
* Applies the `setWafUpgradeSeen` action
* Sets the `Popover` component to display conditional to the presence of a `jetpack_waf_upgrade_seen` user meta flag that is set when:
   * Clicking the `close` icon in the `Popover`
   * Clicking the `Got it` button in the `Popover`
   * Toggling the Automatic rules on for the first time
* Sets the badge to display conditional to the timestamp of the `jetpack_waf_upgrade_badge_timestamp` user meta, which is set when either the actions listed above are performed, is less than 7 days from the current period

#### Other information:


#### Jetpack product discussion


- 1202232924268038-as-1203286905507133

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:






- Checkout this branch
- Load your Jurassic Tube, with no upgrade applied
- Activate Protect
- Proceed to Protect admin page and select the Firewall tab
- Verify that the Automatic rules section is disabled
- Upgrade to a scan-level plan and return to the Firewall admin page
- Verify that both the upgrade `Popover` component and the badge are present
- Perform one of the listed actions to initiate the dismissal of the Popover and to set the timestamp for the badge
- Verify that the badge remains but that Popover is permanently dismissed
   - As an added testing measure, update the `Jetpack_Protect::get_waf_upgrade_badge_display_status` method `$seven_days` variable to `strtotime( '-10 minutes' )` in `class-jetpack-protect.php` and ensure the badge no longer displays after 10 minutes has since setting it.
- Ensure no regression in app functionality

#### Screenshots:
<img width="1040" alt="Screen Shot 2022-12-07 at 09 18 35" src="https://user-images.githubusercontent.com/43220201/206246859-6c5b6356-05fc-41f9-a9e3-d54728d5e585.png">
